src/serial_port.rs

```diff
@@ -268,3 +269,18 @@ where
         }
     }
 }
+
+impl<B, RS, WS> fmt::Write for SerialPort<'_, B, RS, WS>
+where
+    B: UsbBus,
+    RS: BorrowMut<[u8]>,
+    WS: BorrowMut<[u8]>,
+{
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        match <SerialPort<'_, B, RS, WS>>::write(self, s.as_bytes()) {
+            Ok(0) | Err(UsbError::WouldBlock) => Ok(()),
```

Comment:
From `core::fmt::Write` docs:
> This method can only succeed if the entire string slice was successfully written, and this method will not return until all data has been written or an error occurs.

So you have to either:
* return an error when `SerialPort::write` fails - makes this implementation pretty useless
* block until all data has been written - impossible with USB because you have to process EP0 requests at the same time

Reply:
I could return `UsbError::WouldBlock`, which would be an ok "error". I know it's not exactly the heart of the docs intent, but it makes this useful and usable.

Reply:
Why it would be "ok"? How user should recover from this error?

Reply:
Ah, I was thinking the error handling could move to the users code, but I forgot that this is called under the hood. Do you have any ideas? probably isn't doable for the current set of traits available in `core`. Something like [`std::io::Write`](https://doc.rust-lang.org/std/io/trait.Write.html) seems more applicable.

Reply:
This crate is `#![no_std]`, so there is no `std::io::Write` available (and no counterpart in `core`).

The only idea I have is to implement `fmt::Write` on top of the user-defined USB device class, so it can block by polling the whole device at the same time.